circuits/lib/query/credentialAtomicQueryMTPV2.circom

```diff
@@ -0,0 +1,161 @@
+    userIdOwnership.userAuthClaimNonRevMtpNoAux <== userAuthClaimNonRevMtpNoAux;
+    userIdOwnership.userAuthClaimNonRevMtpAuxHv <== userAuthClaimNonRevMtpAuxHv;
+    userIdOwnership.userAuthClaimNonRevMtpAuxHi <== userAuthClaimNonRevMtpAuxHi;
+
+    userIdOwnership.userRootsTreeRoot <== userRootsTreeRoot; // currentHolderStateClaimsRootsTreeRoot
+
+    userIdOwnership.challenge <== challenge;
+    userIdOwnership.challengeSignatureR8x <== challengeSignatureR8x;
+    userIdOwnership.challengeSignatureR8y <== challengeSignatureR8y;
+    userIdOwnership.challengeSignatureS <== challengeSignatureS;
+
+    userIdOwnership.userState <== userState;
+
+    // verify issuerClaim issued and not revoked
+    component vci = verifyClaimIssuanceNonRev(IssuerLevels);
+    for (var i=0; i<8; i++) { vci.claim[i] <== issuerClaim[i]; }
+    for (var i=0; i<IssuerLevels; i++) { vci.claimIssuanceMtp[i] <== issuerClaimMtp[i]; }
+    vci.claimIssuanceClaimsTreeRoot <== issuerClaimClaimsTreeRoot;
+    vci.claimIssuanceRevTreeRoot <== issuerClaimRevTreeRoot;
+    vci.claimIssuanceRootsTreeRoot <== issuerClaimRootsTreeRoot;
+    vci.claimIssuanceIdenState <== issuerClaimIdenState;
+
+    // non revocation status
+    for (var i=0; i<IssuerLevels; i++) { vci.claimNonRevMtp[i] <== issuerClaimNonRevMtp[i]; }
+    vci.claimNonRevMtpNoAux <== issuerClaimNonRevMtpNoAux;
+    vci.claimNonRevMtpAuxHi <== issuerClaimNonRevMtpAuxHi;
+    vci.claimNonRevMtpAuxHv <== issuerClaimNonRevMtpAuxHv;
+    vci.claimNonRevIssuerClaimsTreeRoot <== issuerClaimNonRevClaimsTreeRoot;
+    vci.claimNonRevIssuerRevTreeRoot <== issuerClaimNonRevRevTreeRoot;
+    vci.claimNonRevIssuerRootsTreeRoot <== issuerClaimNonRevRootsTreeRoot;
+    vci.claimNonRevIssuerState <== issuerClaimNonRevState;
+
+    // Check issuerClaim is issued to provided identity
+    component claimIdCheck = verifyCredentialSubject();
+    for (var i=0; i<8; i++) { claimIdCheck.claim[i] <== issuerClaim[i]; }
+    claimIdCheck.id <== userID;
+
+    // Verify issuerClaim schema
+    component claimSchemaCheck = verifyCredentialSchema();
+    for (var i=0; i<8; i++) { claimSchemaCheck.claim[i] <== issuerClaim[i]; }
+    claimSchemaCheck.schema <== claimSchema;
+
+    // verify issuerClaim expiration time
+    component claimExpirationCheck = verifyExpirationTime();
+    for (var i=0; i<8; i++) { claimExpirationCheck.claim[i] <== issuerClaim[i]; }
+    claimExpirationCheck.timestamp <== timestamp;
+
+    // Query
+    component getClaimValue = getValueByIndex();
+    for (var i=0; i<8; i++) { getClaimValue.claim[i] <== issuerClaim[i]; }
+    getClaimValue.index <== slotIndex;
+
+    component q = Query(valueArraySize);
+    q.in <== getClaimValue.value;
+    q.operator <== operator;
+    component poseidonUtil = PoseidonUtil(valueArraySize);
+    for(var i = 0; i < valueArraySize; i++) {  
+        q.value[i] <== value[i];
+        poseidonUtil.in[i] <== value[i];
+    }
+    valueHash <== poseidonUtil.out;
+    q.out === 1;
```

Comment:
Minor: better put this constraint just after the code 
```
    component q = Query(valueArraySize);
    q.in <== getClaimValue.value;
    q.operator <== operator;
```
It's easier to read when we keep it together